pulp/apis/highs_api.py

```diff
@@ -0,0 +1,163 @@
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+# CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE."""
+
+# Modified by Sam Mathew (@samiit on Github)
+# Users would need to install HiGHS on their machine and provide the path to the executable. Please look at this thread: https://github.com/ERGO-Code/HiGHS/issues/527#issuecomment-894852288
+# More instructions on: https://www.highs.dev
+
+from .core import LpSolver_CMD, subprocess, PulpSolverError
+import os, sys
+from .. import constants
+import warnings
+
+
+class HiGHS_CMD(LpSolver_CMD):
+    """The HiGHS_CMD solver"""
+    name = 'HiGHS_CMD'
+
+    def __init__(self, path=None, keepFiles=False, mip=True, msg=True, options=None,  timeLimit=None):
+        """
+        :param bool mip: if False, assume LP even if integer variables
+        :param bool msg: if False, no log is shown
+        :param float timeLimit: maximum time for solver (in seconds)
+        :param list options: list of additional options to pass to solver
+        :param bool keepFiles: if True, files are saved in the current directory and not deleted after solving
+        :param str path: path to the solver binary
+        """
+        LpSolver_CMD.__init__(self, mip=mip, msg=msg, timeLimit=timeLimit,
+                              options=options, path=path, keepFiles=keepFiles)
+
+    def defaultPath(self):
+        return self.executableExtension("")
+
+    def available(self):
+        """True if the solver is available"""
+        return self.executable(self.path)
+
+    def actualSolve(self, lp):
+        """Solve a well formulated lp problem"""
+        if not self.executable(self.path):
+            raise PulpSolverError("PuLP: cannot execute " + self.path)
+        tmpMps, tmpSol, tmpOptions, tmpLog = self.create_tmp_files(lp.name, 'mps', 'sol', 'HiGHS', 'HiGHS_log')
+        write_lines = [f"solution_file = {tmpSol}\n", 
+                       "write_solution_to_file = true\n", 
+                       "write_solution_pretty = true\n"]
+        with open(tmpOptions, "w") as fp: 
+            fp.writelines(write_lines)
+                    
+        if lp.sense == constants.LpMaximize:
+            # we swap the objectives
+            # because it does not handle maximization.
+            warnings.warn('HiGHS_CMD does not currently allow maximization, '
+                          'we will minimize the inverse of the objective function.')
+            lp += -lp.objective
+        lp.checkDuplicateVars()
+        lp.checkLengthVars(52)
+        lp.writeMPS(tmpMps, mpsSense=lp.sense)
```

Comment:
Should the `mpsSense` here just be `LpMinimize`? We negated the objective if it was maximize.

Reply:
You're right. But maybe HiGHS ignores it anyway, but correct me if I am wrong. In any case, I made the changes. Kindly review the latest commit.